frontend/src/components/productGrid/ModelDropdownComponent.js

```diff
@@ -19,26 +19,20 @@ const ModelDropdownComponent = (props) => {
 
     // eslint-disable-next-line
     const [selectedProducts, setSelectedProducts] = useContext(PrivateSectionContext);
+
     // set the initial values for the dropdown list derived from getModels
     const variables = getModels(productID);
-    // console.log('variables');
-    // console.log(variables);
-    const [selected, setSelected] = useState('Select a model');
-    //checking if the variable list is empty
 
-    /* useEffect(() => {
-        async function getProductModels() {
-            const variables = getModels(productID);
-            setVariableList(variables);
-            console.log(variables);
-        }
-        getProductModels();
-    }, []); */
+    const [selected, setSelected] = useState('Select a model');
 
+    //checking if the variable list is empty
     if (variables === [] || variables === undefined || variables === null) {
         return <LoadingComponent />;
     }
-    // else:
+
+    if (variables.length === 1) {
+        return <h3>{variables[0].modelName}</h3>;
+    }
```

Comment:
"too good" or "so good"? look who is correcting you! 😄 always wanted to correct you. 

I haven't worked yet on the UI of this Issue. but you are right.
 isn't it better when it has one model then, just be a text. doesn't make sense for me to have a button here which is not clickable. furthermore, it will be also separated from the ones that have multiple models.  

but for sure the issue with long model names should be fixed.

Reply:
about the case in which no variable is available. what should be done? I guess then it can't be navigated to the details page.

Reply:
Sorry for the late reply: But I guess we talked about that case yesterday. If there are no models available the product name is the model name.

You mentioned that in that case we would display the same name two times. So should we in that case not display the button maybe? If we would do that we need to align the grid items at the top not in the middle of a row.